unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/VoiceChat/DCLVoiceChatController.cs

```diff
@@ -61,50 +69,38 @@ private void IsVoiceChatRecordingChanged(KeyValuePair<bool, bool> current, KeyVa
             if (!DataStore.i.voiceChat.isJoinedToVoiceChat.Get())
                 return;
 
-            CreateSocialAnalyticsIfNeeded();
-
             if (current.Key)
-            {
-                if (!isVoiceChatToggledOn)
-                {
-                    Interface.WebInterface.SendSetVoiceChatRecording(true);
-                    SendFirstTimeMetricIfNeeded();
-                    voiceMessageStartTime = Time.realtimeSinceStartup;
-                }
-            }
+                StartRecording();
             else
-            {
-                Interface.WebInterface.SendSetVoiceChatRecording(false);
+                StopRecording(current.Value);
+        }
 
-                socialAnalytics.SendVoiceMessage(
-                    Time.realtimeSinceStartup - voiceMessageStartTime, 
-                    (current.Value || isVoiceChatToggledOn) ? VoiceMessageSource.Shortcut : VoiceMessageSource.Button, 
-                    userProfileWebInterfaceBridge.GetOwn().userId);
+        private void StartRecording()
+        {
+            if (isRecording) return;
 
-                isVoiceChatToggledOn = false;
-            }
+            WebInterface.SendSetVoiceChatRecording(true);
+
+            CreateSocialAnalyticsIfNeeded();
+            SendFirstTimeMetricIfNeeded();
+            voiceMessageStartTime = Time.realtimeSinceStartup;
+
+            isRecording = true;
         }
 
-        private void ToggleVoiceChatRecording()
+        private void StopRecording(bool usedShortcut)
         {
-            if (!DataStore.i.voiceChat.isJoinedToVoiceChat.Get())
-                return;
+            if (!isRecording) return;
```

Comment:
Could by any reason this state be in desync from kernel? Is `WebInterface.SendSetVoiceChatRecording(true);` getting called anywhere else? Perhaps we can always call `WebInterface.SendSetVoiceChatRecording(false);` in this method, ignoring the current pre-condition?

Reply:
I was not completely sure the `.OnChage` would propagate correctly when losing focus, so I added this precondition as a fail-safe just in case it got called when regaining focus. Not to call twice the `StopRecording` as well as sending the analytic. 

Anyways, we found another desync condition in kernel due to a Promise related to initializing the mic. Im gonna ask you another review once I ve finished validating that one.

Sorry for the noise